streams/src/test/java/org/apache/kafka/streams/kstream/internals/SlidingWindowedCogroupedKStreamImplTest.java

```diff
@@ -156,7 +156,7 @@ public void slidingWindowAggregateStreamsTest() {
             final TestInputTopic<String, String> testInputTopic = driver.createInputTopic(
                     TOPIC, new StringSerializer(), new StringSerializer());
             final TestOutputTopic<Windowed<String>, String> testOutputTopic = driver.createOutputTopic(
-                    OUTPUT, new TimeWindowedDeserializer<>(new StringDeserializer()), new StringDeserializer());
+                    OUTPUT, new TimeWindowedDeserializer<>(new StringDeserializer(), Long.MAX_VALUE), new StringDeserializer());
```

Comment:
Seems we setup the window with `500ms`:
```
SlidingWindows.withTimeDifferenceAndGrace(ofMillis(500L),...)
```
Thus we should pass in the same value instead of `MAX_VAVLUE`.

We should also change `TestRecord<String,String>` below to `TestRecorded<Windowed<String>,String>`

Reply:
I've changed the code to pass `500L`, but the `TestRecord` change causes compilation errors since we explicitly use a `nonWindowedRecord`:

> results.add(nonWindowedRecord);

Can we address that separately? I think we can focus on fixing the deprecation warning in this PR, if that's ok with you.